### PR TITLE
[FIX] budget_control: fix check bcs status

### PR DESCRIPTION
**Pull request by ps-tubtim**
Fix function `_check_budget_control_status`.

- In case the analytic account has the budget control sheets more than one by 1 control and 1 cancel, I cannot reserve budget because the system found `budget_not_controlled`.

```
budget_not_controlled = budget_controls.filtered_domain(
            [("state", "!=", "done")]
        )
        if budget_not_controlled:
            names = budget_not_controlled.mapped("analytic_account_id.display_name")
```

CC: @kittiu

### CODE REVIEW

File: budget_control/models/analytic_account.py [hunk 1]
```diff
@@ -205,11 +205,13 @@ def _check_budget_control_status(self, budget_period_id=False):
...
             )
-        budget_not_controlled = budget_controls.filtered_domain(
-            [("state", "!=", "done")]
-        )
-        if budget_not_controlled:
-            names = budget_not_controlled.mapped("analytic_account_id.display_name")
+        # Find analytics has no controlled budget control sheet
+        budget_controlled = budget_controls.filtered_domain([("state", "=", "done")])
+        cbc_analytic_ids = budget_controlled.mapped("analytic_account_id").ids
+        no_cbc_analytic_ids = list(set(self.ids) - set(cbc_analytic_ids))
```
Comment (kittiu) [inline budget_control/models/analytic_account.py]: Can we oper with objects directly? So you don't need to browse again. For better performance.

Comment (ps-tubtim) [reply budget_control/models/analytic_account.py]: Thank you, I fixed it.

File: budget_control/models/analytic_account.py [hunk 2]
```diff
@@ -195,27 +195,27 @@ def _check_budget_control_status(self, budget_period_id=False):
...
+                % names
             )
-        budget_not_controlled = budget_controls.filtered_domain(
-            [("state", "!=", "done")]
-        )
-        if budget_not_controlled:
-            names = budget_not_controlled.mapped("analytic_account_id.display_name")
+        # Find analytics has no controlled budget control sheet
+        budget_controlled = budget_controls.filtered_domain([("state", "=", "done")])
+        cbc_analytic_ids = budget_controlled.mapped("analytic_account_id")
```
Comment (kittiu) [inline budget_control/models/analytic_account.py]: For objects, can we don't use `_ids` for variables.

Comment (ps-tubtim) [reply budget_control/models/analytic_account.py]: I already updated it.

**Review by kittiu — APPROVED** [review]

**Pull request merged**